vircadia/vircadia-web-sdk
# Anonymous Web SDK connection to domain server

`DomainServer` API additions:
- `connect(url)`: WebSocket URL for the Doman Server's WebRTC signaling WebSocket port, e.g., `ws://127.0.0.1:40102`.
- `disconnect()`
- `location`: Read-only property returning the current location URL.
- `state`: Read-only numeric property returning the current connection state.
- `stateToString(state)`: Returns a string representation of the given connection state.
- `DISCONNECTED` etc.: Read-only properties of connection state values.
- `refusalInfo`: Read-only property with description of connection `REFUSED` state.
- `errorInfo`: Read-only property with description of connection `ERROR` state.
- `onStateChanged`: Write-only property that sets a callback function to be called when the connection state changes.

Example Web app provided in `./example` directory.

Builds on top of PR https://github.com/vircadia/vircadia-web-sdk/pull/20.
Works in conjunction with Vircadia PR https://github.com/vircadia/vircadia/pull/1300.

Merge sequence:
- [x] PR https://github.com/vircadia/vircadia-web-sdk/pull/20.
- [x] CR, QA.

#### QA

With the domain server running on localhost:
- Enable anonymous connections and restart.

In the directory where the PR files are checked out:
` `npm install` - To set up the the dev & test environment.
- `npm run lint` - Should complete with no errors.
- `npm run build` - Should complete with no errors.
- `npm run test` - All  tests should pass. 
    Run the tests against the domain server in https://github.com/vircadia/vircadia/pull/1300.
    Check that the domain-server.exe process doesn't crash and restart during the tests.
    Known issue: "Jest did not exit one second after the test run has completed." See: https://github.com/vircadia/vircadia-web-sdk/issues/21
- `npm run devdoc` : Docs should be created in `./docs/dev/` directory.
- `npm run sdkdoc` : Docs should be created in `./docs/sdk/` directory.
- Set up and run the example Web app per the README.md in the `./example` directory.
  Note: If anonymous connection isn't permitted, the app will show a "REFUSED" state.
  Known issue: State reported by `DomainServer` isn't "ERROR" if location used is invalid.
